crates/air/src/commands/format.rs

```diff
@@ -1,64 +1,312 @@
+use std::fmt::Display;
+use std::fmt::Formatter;
+use std::io;
+use std::io::stdout;
+use std::io::Write;
 use std::path::PathBuf;
 
+use air_fs::relativize_path;
 use air_r_formatter::context::RFormatOptions;
 use air_r_parser::RParserOptions;
+use ignore::DirEntry;
+use itertools::Either;
+use itertools::Itertools;
 use line_ending::LineEnding;
+use thiserror::Error;
 
 use crate::args::FormatCommand;
 use crate::ExitStatus;
 
 pub(crate) fn format(command: FormatCommand) -> anyhow::Result<ExitStatus> {
-    format_file(&command.file)
+    let mode = FormatMode::from_command(&command);
+    let paths = resolve_paths(&command.paths);
+
+    let (results, errors): (Vec<_>, Vec<_>) = paths
+        .into_iter()
+        .map(|path| match path {
+            Ok(path) => format_file(path, mode),
+            Err(err) => Err(err.into()),
+        })
+        .partition_map(|result| match result {
+            Ok(result) => Either::Left(result),
+            Err(err) => Either::Right(err),
+        });
+
+    for error in &errors {
+        // TODO: Hook up a tracing subscriber!
+        tracing::error!("{error}");
+    }
```

Comment:
Errors are collected along the way for each path and are emitted at top level here. It's silent right now, but once we hook up a subscriber it will look quite nice as it reports the path and the problem for each error